pkg/migrate/cstor/pool.go

```diff
@@ -0,0 +1,431 @@
+		CStorPoolInstances(c.OpenebsNamespace).
+		Update(cspiObj)
+	if err != nil {
+		return err
+	}
+	err = retry.
+		Times(60).
+		Wait(5 * time.Second).
+		Try(func(attempt uint) error {
+			cspiObj, err1 = c.OpenebsClientset.CstorV1().
+				CStorPoolInstances(c.OpenebsNamespace).
+				Get(cspiObj.Name, metav1.GetOptions{})
+			if err1 != nil {
+				return err1
+			}
+			if cspiObj.Status.Phase != "ONLINE" {
+				return errors.Errorf("failed to verify cspi %s phase expected: Healthy got: %s",
+					cspiObj.Name, cspiObj.Status.Phase)
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+	err = c.updateCVRsLabels(cspiObj)
+	if err != nil {
+		return err
+	}
+	// remove the finalizers from csp object for cleanup
+	// as pool pod is no longer running to remove them.
+	return removeCSPFinalizers(cspObj)
+}
+
+func removeCSPFinalizers(cspObj *apis.CStorPool) error {
+	cspClient := csp.KubeClient()
+	newCSP := cspObj.DeepCopy()
+	newCSP.Finalizers = []string{}
+	patchData, err := GetPatchData(cspObj, newCSP)
+	if err != nil {
+		return err
+	}
+	_, err = cspClient.Patch(
+		cspObj.Name,
+		k8stypes.MergePatchType,
+		patchData,
+	)
+	return err
+}
+
+// get csp for cspi on the basis of cspLabel, which is the combination of
+// hostname label on which cspi came up and the spc label.
+func getCSP(cspLabel string) (*apis.CStorPool, error) {
+	cspClient := csp.KubeClient()
+	cspList, err := cspClient.List(metav1.ListOptions{
+		LabelSelector: cspLabel,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(cspList.Items) != 1 {
+		return nil, fmt.Errorf("Invalid number of pools on one node: %v", cspList.Items)
+	}
+	cspObj := cspList.Items[0]
+	return &cspObj, nil
+}
+
+// The old pool pod should be scaled down before the new cspi pod reconcile is
+// enabled to avoid importing the pool at two places at the same time.
+func (c *CSPCMigrator) scaleDownDeployment(cspObj *apis.CStorPool, openebsNamespace string) error {
+	klog.Infof("Scaling down deployemnt %s", cspObj.Name)
+	cspDeployList, err := c.KubeClientset.AppsV1().
+		Deployments(openebsNamespace).List(
+		metav1.ListOptions{
+			LabelSelector: "openebs.io/cstor-pool=" + cspObj.Name,
+		})
+	if err != nil {
+		return err
+	}
+	if len(cspDeployList.Items) != 1 {
+		return errors.Errorf("invalid number of csp deployment found for %s: %d", cspObj.Name, len(cspDeployList.Items))
+	}
+	_, err = c.KubeClientset.AppsV1().Deployments(openebsNamespace).
+		Patch(
+			cspDeployList.Items[0].Name,
+			k8stypes.StrategicMergePatchType,
+			[]byte(replicaPatch),
+		)
+	if err != nil {
+		return err
+	}
+	err = retry.
+		Times(60).
+		Wait(5 * time.Second).
+		Try(func(attempt uint) error {
+			cspPods, err1 := c.KubeClientset.CoreV1().
+				Pods(openebsNamespace).
+				List(metav1.ListOptions{
+					LabelSelector: "openebs.io/cstor-pool=" + cspObj.Name,
+				})
+			if err1 != nil {
+				return errors.Wrapf(err1, "failed to get csp deploy")
+			}
+			if len(cspPods.Items) != 0 {
```

Comment:
Good to print a log saying waiting for pool deployment to scale down.  Similarly in other places also.. Ignore this comment if retry is already printing it.